datamodel.customer-survey.xml

```diff
@@ -1922,6 +1925,59 @@ EOF
 				throw new Exception(Dict::S('Survey-email-notsent').': '.implode(', ', $aErrors));
 		}
 	}]]></code>
+        </method>
+        <method id="ApplyParamsToContent">
+          <comment>/**
+	 * @param string $sContent
+	 * @param Contact $oContact
+	 * @param ?string $sQuizUrl
+	 * @param boolean $bAutoAppendLink see corresponding param in {@see AppendLinkToContent}
+	 *
+	 * @return string Content with placeholders replaced.
+	 */</comment>
+          <static>false</static>
+          <access>protected</access>
+          <type>Custom</type>
+          <code><![CDATA[	protected function ApplyParamsToContent($sContent, $oContact, $sQuizUrl, $bAutoAppendLink)
+{
+  $aParams = [];
+  $aParams['recipient->object()'] = $oContact;
+  $aParams['quiz_url'] = $sQuizUrl;
+  $aParams['this->object()'] = $this;
+
+  $sContent = $this->AppendLinkToContent($bAutoAppendLink, $sContent, $sQuizUrl);
+
+  $sContent = MetaModel::ApplyParams($sContent, $aParams);
+
+  return $sContent;
+}
+]]></code>
+        </method>
+        <method id="AppendLinkToContent">
+          <comment>/**
+	 * @param boolean $bAutoAppendLink if false will always return the content, otherwise will append a quiz link if no url placeholder found in content
+	 * @param string $sContent
+	 * @param ?string $sQuizUrl
+	 *
+	 * @return string
+*/</comment>
+          <static>true</static>
+          <access>public</access>
+          <type>Custom</type>
+          <code><![CDATA[	public static function AppendLinkToContent($bAutoAppendLink, $sContent, $sQuizUrl)
+{
+    if (false === $bAutoAppendLink) {
+        return $sContent;
+    }
```

Comment:
Again, it doesn't make sense to call this method if not wanted to append a link..

Reply:
Right, I agree.
There are some adaptations to do... @jbostoen don't hesitate to change it.
Otherwise it will wait after I'm back from holidays :/

Reply:
Should be fixed in 45c45820 !